crates/ruff/src/rules/pycodestyle/rules/missing_whitespace.rs

```diff
@@ -40,21 +36,26 @@ pub fn missing_whitespace(
     indent_level: usize,
 ) -> Vec<Diagnostic> {
     let mut diagnostics = vec![];
-    for (idx, char) in line.chars().enumerate() {
-        if idx + 1 == line.len() {
-            break;
+
+    let mut num_lsqb = 0;
+    let mut num_rsqb = 0;
+    let mut prev_lsqb = None;
+    let mut prev_lbrace = None;
+    for (idx, (char, next_char)) in line.chars().tuple_windows().enumerate() {
+        if char == '[' {
```

Comment:
Not. I like to use match statements if I have many branches matching on constant values. It gives you compile time evaluation that no teo branches are overlapping (and can be shorter than changing char == comparisons)

Reply:
(I think a lot of this code is a line-by-line port of pycodestyle, for correctness purposes as we were trying to get the test suite passing. Doesn't mean it has to stay that way :))